### PR TITLE
[PATCH v3] test: pktio_ordered: add option to enable promisc mode

### PR DESCRIPTION
**Pull request by v01fi**
Add command line option '-P, --promisc_mode' to pktio ordered test
application, which allow to enable promiscuous mode for the devices that
supports it.

Signed-off-by: Volodymyr Fialko <vfialko@marvell.com>
Reviewed-by: Matias Elo <matias.elo@nokia.com>

### CODE REVIEW

File: test/performance/odp_pktio_ordered.c [hunk 1]
```diff
@@ -1025,7 +1046,10 @@ static void print_info(char *progname, appl_args_t *appl_args)
 	       progname, appl_args->if_count);
 	for (i = 0; i < appl_args->if_count; ++i)
 		printf(" %s", appl_args->if_names[i]);
-	printf("\n\n");
+	printf("\n");
+	printf("Promisc mode:    %s\n", appl_args->promisc_mode ?
```
Comment (MatiasElo) [inline test/performance/odp_pktio_ordered.c:1050]: Currently the application parameters are printed in two places, here and in `main():1291`. You could move also the rest of the prints here, so everything would be printed nicely in one place.

**Review by MatiasElo — COMMENTED** [review]
Couple minor comments, but otherwise looks good. Please change the commit subject line to: `test: pktio_ordered: add option to enable promisc mode`. The option name (e.g. `-p`) could also be mentioned in the commit log.

File: test/performance/odp_pktio_ordered.c (hunk 1 above)
Comment (v01fi) [reply test/performance/odp_pktio_ordered.c]: Moved.

File: test/performance/odp_pktio_ordered.c [hunk 2]
```diff
@@ -823,6 +838,7 @@ static void usage(char *progname)
 	       "  -a, --accuracy <number>     Statistics print interval in seconds\n"
 	       "                              (default is 1 second).\n"
 	       "  -d, --dst_addr  Destination addresses (comma-separated, no spaces)\n"
+	       "  -P, --promisc_mode          Enable promiscuous mode.\n"
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply test/performance/odp_pktio_ordered.c:841]: Using `-P` for consistency is OK.

File: test/performance/odp_pktio_ordered.c [hunk 3]
```diff
@@ -1025,7 +1046,18 @@ static void print_info(char *progname, appl_args_t *appl_args)
...
+	       "Mode:            %s\n"
+	       "Flows:           %d\n"
+	       "Extra rounds:    %d\n"
+	       "Promisc mode:    %s\n\n", appl_args->num_rx_q,
+	       (appl_args->in_mode == SCHED_ATOMIC) ? "PKTIN_SCHED_ATOMIC" :
+	       (appl_args->in_mode == SCHED_PARALLEL ? "PKTIN_SCHED_PARALLEL" :
+	       "PKTIN_SCHED_ORDERED"), appl_args->num_flows,
+	       appl_args->extra_rounds, appl_args->promisc_mode ?
+	       "enabled" : "disabled");
 	fflush(NULL);
```
Comment (MatiasElo) [inline test/performance/odp_pktio_ordered.c:1061]: The current output
```
Running ODP appl: "odp_pktio_ordered"
-----------------
IF-count:        1
Using IFs:       loop
Application parameters
----------------------
Input queues:    1
Mode:            PKTIN_SCHED_ORDERED
Flows:           12
Extra rounds:    15
Promisc mode:    disabled

Num worker threads: 1
First CPU:          47
CPU mask:           0x800000000000
```
could be cleaned a bit to something like:
```
odp_pktio_ordered options
-------------------------
IF-count:           1
Using IFs:          loop
Input queues:       1
Mode:               PKTIN_SCHED_ORDERED
Flows:              12
Extra rounds:       15
Promisc mode:       disabled
Num worker threads: 1
First CPU:          47
CPU mask:           0x800000000000
```

**Review by MatiasElo — COMMENTED** [review]
One "beautification" comment. Please add my reviewed-by tag in the next version.

**Review by MatiasElo — APPROVED** [review]

**Pull request merged**